public/app/features/dashboard/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { PureComponent, createRef } from 'react';
+import moment from 'moment';
+import {
+  TimeRange,
+  TimeOptions,
+  TimeOption,
+  SelectOptionItem,
+  ClickOutsideWrapper,
+  SelectButton,
+  HeadlessSelect,
+} from '@grafana/ui';
+import Popper from '@grafana/ui/src/components/Tooltip/Popper';
+
+import { TimePickerOptionGroup } from './TimePickerOptionGroup';
+import { TimePickerPopOver } from './TimePickerPopOver';
+import { mapTimeOptionToTimeRange, mapTimeRangeToRangeString } from './time';
+
+export interface Props {
+  value: TimeRange;
+  isTimezoneUtc: boolean;
+  popOverTimeOptions: TimeOptions;
+  selectTimeOptions: TimeOption[];
+  timezone?: string;
+  onChange: (timeRange: TimeRange) => void;
+  onMoveBackward: () => void;
+  onMoveForward: () => void;
+  onZoom: () => void;
+}
+
+export interface State {
+  isPopOverOpen: boolean;
+  isSelectOpen: boolean;
+  isSmallScreen: boolean;
+}
+
+export class TimePicker extends PureComponent<Props, State> {
+  pickerTriggerRef = createRef<HTMLDivElement>();
+  state = { isSelectOpen: false, isPopOverOpen: false, isSmallScreen: false };
+
+  componentWillMount() {
+    this.setIsSmallScreen();
+    window.addEventListener('resize', this.setIsSmallScreen);
```

Comment:
Would be good to throttle as it triggers a lot of `setState calls`

Reply:
Note to myself -> `useWindowSize` hook in the future?

Reply:
Good catch!

Reply:
or add this to a redux state, there are a few pages that are interested in the window size already (DashboardGrid) and Explore, the tricky thing there is that they are also affected by hiding of the sidemenu so to capture that we they are using AutoSizer (and react-sizeme for DashboardGrid), but think that is not required actually, if we published a redux action on hiding/showing sidemenu, we could have two props in a redux "window" store like. windowWidth, & pageWidth or something like that. We will have one location for listening to resize events (that then publishes a redux action) 

thoughts? does not need to be part of this PR, but if it sounds like a good idea I can create an issue.

Reply:
I'm all for that suggestion

Reply:
Have you created a separate issue for this @torkelo or should I?

Reply:
https://github.com/grafana/grafana/issues/15403